docs/middlewares.md

```diff
@@ -90,6 +90,11 @@ Sets headers in `after` and `onError` phases.
 - `headers` (string) (optional): value to put in Access-Control-Allow-Headers (default: `null`)
 - `credentials` (bool) (optional): if true, sets the `Access-Control-Allow-Origin` as request header `Origin`, if present (default `false`)
 
+NOTES:
+
+- When used with `httpErrorHandler` middleware, ensure to use the `httpErrorHandler` middleware before `cors` to avoid 
+the loss of CORS headers 
```

Comment:
@gsingh1 you need to explain `the loss of CORS headers`, e.g. `if another middleware doesn't handle and swallow the error, then it'll bubble all the way to the top and terminate the Lambda invocation with an error. In this case API Gateway would return a default 502 response, and the CORS headers would be lost. To prevent this, you should use the httpErrorHandler middleware before the cors middleware like this:
<insert code example here>`

Reply:
Updated explanation and added code sample